polling/frequent/activities.py

```diff
@@ -17,6 +19,17 @@ async def compose_greeting(input: ComposeGreetingInput) -> str:
     while True:
         try:
             result = test_service.get_service_result(input)
+            activity.logger.info(f"Exiting activity ${result}")
             return result
-        except Exception:
+        except Exception as e:
+            # swallow exception since service is down
+            activity.logger.error(e)
+
+        try:
             activity.heartbeat("Invoking activity")
+        except asyncio.CancelledError as exception:
+            # activity was either cancelled or workflow was completed or worker shut down
+            # if you need to clean up you can catch this. Here we are just reraising exception
+            raise
```

Comment:
Heartbeat does not raise an exception. This exception would occur from the get service result or sleep call

Reply:
OK that confuses me. I reimplemented with whole thing in CancelledError because I still think it matters